addons/hr/hr.py

```diff
@@ -317,8 +317,7 @@ def name_get(self, cr, uid, ids, context=None):
         return res
 
     def create(self, cr, uid, vals, context=None):
-        if context is None:
-            context = {}
+        context = dict(context) if context is not None else {}
         context['mail_create_nosubscribe'] = True
```

Comment:
honnestly I don't care, as long as the bug is fixed. Using the patch I
provided will avoid conflict in OCB, but this will be easy to fix.


2015-12-27 2:20 GMT+01:00 Pedro M. Baeza <notifications@github.com>:

> In addons/hr/hr.py
> <https://github.com/odoo/odoo/pull/9740#discussion_r48451024>:
>
> > @@ -317,8 +317,7 @@ def name_get(self, cr, uid, ids, context=None):
> >          return res
> >
> >      def create(self, cr, uid, vals, context=None):
> > -        if context is None:
> > -            context = {}
> > +        context = dict(context) if context is not None else {}
> >          context['mail_create_nosubscribe'] = True
>
> Ah, OK. Of course it's only a matter of personal preference. Both options
> are valid, although the proposed one is friendlier with coverage.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/odoo/odoo/pull/9740/files#r48451024>.
>



-- 
Alexandre Fayolle
Chef de Projet
Tel : + 33 (0)4 58 48 20 30

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac Cedex
http://www.camptocamp.com